soh/src/overlays/actors/ovl_Bg_Spot03_Taki/z_bg_spot03_taki.c

```diff
@@ -52,11 +53,20 @@ void BgSpot03Taki_Init(Actor* thisx, PlayState* play) {
 
     this->switchFlag = (this->dyna.actor.params & 0x3F);
     DynaPolyActor_Init(&this->dyna, DPM_UNK);
-    CollisionHeader_GetVirtual(&object_spot03_object_Col_000C98, &colHeader);
-    this->dyna.bgId = DynaPoly_SetBgActor(play, &play->colCtx.dyna, &this->dyna.actor, colHeader);
+
+    if (GameInteractor_Should(VB_KEEP_ZORA_WATERFALL_OPEN, false)) {
```

Comment:
You can read up more on it here, but the entire idea behind the VB/hooking system is to leave the exact vanilla codepaths intact whenever possible, and not proliferate the source files with our own branches

https://github.com/HarbourMasters/Shipwright/pull/3755

Additionally, because you are just doing some things in the init, you can likely solve all of this in an `OnActorInit` handler and not touch a single line of the source code.